src/promql/src/planner.rs

```diff
@@ -261,24 +257,34 @@ impl PromPlanner {
                     (None, None) => {
                         let left_input = self.prom_expr_to_plan(*lhs.clone()).await?;
                         let left_field_columns = self.ctx.field_columns.clone();
-                        let left_schema = left_input.schema().clone();
+                        let left_table_ref: OwnedTableReference =
+                            self.ctx.table_name.clone().unwrap_or_default().into();
 
                         let right_input = self.prom_expr_to_plan(*rhs.clone()).await?;
                         let right_field_columns = self.ctx.field_columns.clone();
-                        let right_schema = right_input.schema().clone();
+                        let right_table_ref: OwnedTableReference =
+                            self.ctx.table_name.clone().unwrap_or_default().into();
+
+                        // TODO(ruihang): avoid join if left and right are the same table
```

Comment:
I think we can return an error directly instead of leaving a TODO here.

Reply:
No, it's supported now. This todo is an optimization

Reply:
ok